fetch/cosmwasm_contract/src/access_control.rs

```diff
@@ -58,55 +58,45 @@ impl FromStr for AccessRole {
     }
 }
 
-pub fn access_control<S: Storage>(storage: &mut S) -> PrefixedStorage<S> {
-    return PrefixedStorage::new(ACCESS_CONTROL_KEY, storage);
-}
-
-pub fn access_control_read<S: Storage>(storage: &S) -> ReadonlyPrefixedStorage<S> {
-    return ReadonlyPrefixedStorage::new(ACCESS_CONTROL_KEY, storage);
-}
-
-pub fn ac_have_role<S: Storage>(
-    storage: &S,
-    addr: &HumanAddr,
-    role: &AccessRole,
-) -> StdResult<bool> {
-    let ac_store = access_control_read(storage);
-    let addr_roles = ReadonlyPrefixedStorage::new(addr.as_str().as_bytes(), &ac_store);
-    match addr_roles.get(role.as_bytes()) {
-        Some(_) => Ok(true),
-        None => Ok(false),
+pub fn ac_have_role(storage: &dyn Storage, addr: &HumanAddr, role: &AccessRole) -> StdResult<bool> {
+    let addr_roles = ReadonlyPrefixedStorage::multilevel(
+        storage,
+        &[ACCESS_CONTROL_KEY, addr.as_str().as_bytes()],
```

Comment:
```suggestion
        &[ACCESS_CONTROL_KEY, addr.as_bytes()],
```

The as_str() is redundant